analyses/phylo_beta_diversity/phylo_beta_diversity.py

```diff
@@ -478,3 +498,194 @@ def calculate_phylo_beta_diversity_sorensen(pam, tree):
+    #    present in both the rows and the columns.
+    mtx_headers = {
+        '0': pam.get_row_headers(),  # Row headers
+        '1': pam.get_row_headers()  # Column headers
+    }
+
+    num_sites = pam.data.shape[0]  # Get the number of sites in the PAM
+
+    # Matrices to hold means in returnable format.
+    SES = np.zeros((num_sites, num_sites), dtype=np.float)
+
+    # TODO: Randomize tree, calc. metrics, save running average & SD.
+    init_tree = tree  # just making a copy in case is needed.
+
+    # These lists hold objects for calc. running stats.
+    rs_mat = []
+    # Populate lists with RunningStats objects.
+    for row in range(num_sites):
+        rs_mat.append([])
+        for col in range(num_sites):
+            rs_mat[row].append(rs.RunningStats())
+
+    # Populate rs arrays with beta-diversity metrics.
+    for trial in range(nrand):
+        # Randomize tip labels of tree.
+        # print "****\n", tree.print_plot(), "\n"
+        tree.shuffle_taxa()  # modifies tree in-place.
+        # tree.randomly_assign_taxa() #  modifies tree in-place.
+        # print tree.print_plot(), "\n****\n"
+
+        # Get core metrics.
+        new_core = core_PD_calc(pam,tree)
+
+        # This loop will populate arrays with all beta diversity metrics.
+        for my_row in range(new_core.data.shape[0]):
+            # Pull out the phylogentic core numeric values.
+            my_dat = new_core.data[my_row, 0:4]
+
+            # Get index values for placing into output arrays.
+            my_dim = new_core.get_row_headers()[my_row]
+
+            if metric == 'jtu':
+                rs_mat[my_dim[0]][my_dim[1]].push(
+                    (2*my_dat[0]) / ((2*my_dat[0]) + my_dat[3]))
+
+                rs_mat[my_dim[1]][my_dim[0]].push(
+                    (2*my_dat[0]) / ((2*my_dat[0]) + my_dat[3]))
+
+            elif metric == 'jac':
+                rs_mat[my_dim[0]][my_dim[1]].push(
+                    my_dat[2] / (my_dat[3] + my_dat[2]))
+
+                rs_mat[my_dim[1]][my_dim[0]].push(
+                    my_dat[2] / (my_dat[3] + my_dat[2]))
+
+            elif metric == 'jne':
+                rs_mat[my_dim[0]][my_dim[1]].push(
+                    ((my_dat[1] - my_dat[0]) / (my_dat[3] + my_dat[2])) * (
+                        my_dat[3] / ((2 * my_dat[0]) + my_dat[3])))
+
+                rs_mat[my_dim[1]][my_dim[0]].push(
+                    ((my_dat[1] - my_dat[0]) / (my_dat[3] + my_dat[2])) * (
+                        my_dat[3] / ((2 * my_dat[0]) + my_dat[3])))
+
+	# Populate returnable arrays with av. & SD.
+    for row in range(num_sites):
+        for col in range(num_sites):
+            if row == col:
+                # ses matrix
+                SES[row][col] = 0.
+
+            else:
+                # standardized effect size calculation
+                SES[row][col] = obs.data[row][col] - rs_mat[row][col].mean() / sqrt(rs_mat[row][col].variance())

```

Comment:
Should there be parentheses around "obs.data[row][col] - rs_mat[row][col].mean()"?  Is the equation supposed to be (X - mu) / std dev?  Or X - (mu / std dev)?